go/vt/vtgate/router.go

```diff
@@ -701,57 +704,86 @@ func (rtr *Router) handleGenerate(vcursor *queryExecutor, gen *engine.Generate)
 	return insertid, nil
 }
 
-func (rtr *Router) handlePrimary(vcursor *queryExecutor, vindexKey interface{}, colVindex *vindexes.ColumnVindex, bv map[string]interface{}, rowNum int) (ksid []byte, err error) {
-	if vindexKey == nil {
+func (rtr *Router) handlePrimary(vcursor *queryExecutor, vindexKeys []interface{}, colVindex *vindexes.ColumnVindex, bv map[string]interface{}, allShards []*topodatapb.ShardReference) (keyspaceIDs [][]byte, err error) {
+	if vindexKeys == nil {
```

Comment:
I don't think this is possible. Were you able to get coverage to trigger this path?